packages/expo-updates/android/src/main/java/expo/modules/updates/UpdatesModule.java

```diff
@@ -29,7 +42,152 @@ public void onCreate(ModuleRegistry moduleRegistry) {
     mModuleRegistry = moduleRegistry;
   }
 
+  @Override
+  public Map<String, Object> getConstants() {
+    Map<String, Object> constants = new HashMap<>();
+
+    try {
+      UpdatesController controller = UpdatesController.getInstance();
+      if (controller != null) {
+        constants.put("isEmergencyLaunch", controller.isEmergencyLaunch());
+
+        UpdateEntity launchedUpdate = controller.getLaunchedUpdate();
+        if (launchedUpdate != null) {
+          constants.put("manifestString", launchedUpdate.metadata.toString());
+        }
+
+        Map<AssetEntity, String> localAssetFiles = controller.getLocalAssetFiles();
+        if (localAssetFiles != null) {
+          Map<String, String> localAssets = new HashMap<>();
+          for (AssetEntity asset : localAssetFiles.keySet()) {
+            String localAssetsKey = UpdateUtils.getLocalAssetsKey(asset);
+            if (localAssetsKey != null) {
+              localAssets.put(localAssetsKey, localAssetFiles.get(asset));
+            }
+          }
+          constants.put("localAssets", localAssets);
+        }
+      }
+    } catch (IllegalStateException e) {
+      // do nothing; this is expected in a development client
+    }
+
+    return constants;
+  }
+
+  @ExpoMethod
+  public void reload(final Promise promise) {
+    try {
+      UpdatesController.getInstance().relaunchReactApplication(getContext(), new Launcher.LauncherCallback() {
+        @Override
+        public void onFailure(Exception e) {
+          Log.e(TAG, "Failed to relaunch application", e);
+          promise.reject("ERR_UPDATES_RELOAD", e.getMessage(), e);
+        }
+
+        @Override
+        public void onSuccess() {
+          promise.resolve(null);
+        }
+      });
+    } catch (IllegalStateException e) {
+      promise.reject(
+        "ERR_UPDATES_RELOAD",
+        "The updates module controller has not been properly initialized. If you're using a development client, you cannot use `Updates.reloadAsync`. Otherwise, make sure you have called the native method UpdatesController.initialize()."
+      );
+    }
+  }
+
+  @ExpoMethod
+  public void checkForUpdateAsync(final Promise promise) {
+    try {
+      final UpdatesController controller = UpdatesController.getInstance();
+      FileDownloader.downloadManifest(controller.getManifestUrl(), getContext(), new FileDownloader.ManifestDownloadCallback() {
```

Comment:
This might be just a method in controller which accepts callback. This way we would reference only controller, with no need of knowing that `FileDownloader` exists.